lib/commands/reset_connection.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const Command = require('./command');
+const Packets = require('../packets/index.js');
+
+class ResetConnection extends Command {
+  constructor(callback) {
+    super();
+    this.onResult = callback;
+  }
+
+  start(packet, connection) {
+    const req = new Packets.ResetConnection();
+    connection.writePacket(req.toPacket());
+    return ResetConnection.prototype.resetConnectionResponse;
+  }
+
+  resetConnectionResponse() {
+    if (this.onResult) {
```

Comment:
re statement cache:
1) add private connection helper method connection._initialiseStatementCache({ disableDispose: boolean })

something along the lines:

```js
_initialiseStatementCache({ disableDispose }) {
  if (this._statements) {
    if (disableDispose) {
       this._statements.dispose = null;
    }
    this._statements.clear();
  }
  this._statements = new LRU({
    max: this.config.maxPreparedStatements,
    dispose: function(key, statement) {
      statement.close();
    }
  })
}
```

2. replace line [66 in connection.js](https://github.com/sidorares/node-mysql2/blob/299ed3d95576da72ce71933ea26cb76f8ab21f79/lib/connection.js#L66) with `this._initialiseStatementCache({ disableDispose: false });`

3) in `ResetConnectionCommand.resetConnectionResponse(packet, connection)` add call `connection._initialiseStatementCache({ disableDispose: true })` at the beginning before calling `onResult()`